sdk/textanalytics/ai-text-analytics/src/lro.ts

```diff
@@ -0,0 +1,314 @@
+import { transformAnalyzeBatchResults, transformError } from "./transforms";
+import { HttpMethods } from "@azure/core-rest-pipeline";
+import { TracingClient } from "@azure/core-tracing";
+import { clientName } from "./constants";
+import { logger } from "./logger";
+
+const serializer = createSerializer(Mappers, /* isXml */ false);
+
+const jobStatusOperationSpec: OperationSpec = {
+  httpMethod: "GET",
+  responses: {
+    200: {
+      bodyMapper: Mappers.AnalyzeTextJobState,
+    },
+    default: {
+      bodyMapper: Mappers.ErrorResponse,
+    },
+  },
+  headerParameters: [Parameters.accept],
+  queryParameters: [Parameters.top, Parameters.skip, Parameters.includeStatistics],
+  serializer,
+};
+
+async function getRawResponse<TOptions extends OperationOptions, TResult>(
+  f: (options: TOptions) => Promise<TResult>,
+  options: TOptions
+): Promise<LroResponse<TResult>> {
+  const { onResponse } = options || {};
+  let rawResponse: FullOperationResponse | undefined = undefined;
+  const flatResponse = await f({
+    ...options,
+    onResponse: (response: FullOperationResponse, flatResponseParam: unknown) => {
+      rawResponse = response;
+      onResponse?.(response, flatResponseParam);
+    },
+  });
+  return {
+    flatResponse,
+    rawResponse: {
+      statusCode: rawResponse!.status,
+      headers: rawResponse!.headers.toJSON(),
+      body: rawResponse!.parsedBody,
+    },
+  };
+}
+
+async function sendRequest<TOptions extends OperationOptions>(settings: {
+  client: GeneratedClient;
+  tracing: TracingClient;
+  spec: OperationSpec;
+  spanStr: string;
+  opOptions: TOptions;
+  path: string;
+  httpMethod?: HttpMethods;
+}): Promise<LroResponse<unknown>> {
+  const { client, opOptions, path, spanStr, spec, tracing, httpMethod = "GET" } = settings;
+  return tracing.withSpan(spanStr, opOptions, async (finalOptions: TOptions) => {
+    try {
+      const response = getRawResponse(
+        (options) =>
+          client.sendOperationRequest(
+            { options },
+            {
+              ...spec,
+              path,
+              httpMethod,
+            }
+          ),
+        finalOptions
+      );
+      return response;
+    } catch (e: unknown) {
+      throw transformError(e);
+    }
+  });
+}
+
+/**
+ * @internal
+ */
+export function createSendPollRequest<TOptions extends OperationOptions>(settings: {
+  client: GeneratedClient;
+  tracing: TracingClient;
+  options: TOptions;
+  spanStr: string;
+}): (path: string) => Promise<LroResponse<unknown>> {
+  const { client, options, tracing, spanStr } = settings;
+  return async (path: string): Promise<LroResponse<unknown>> => {
+    return throwTransformErrors(
+      sendRequest({
+        client,
+        opOptions: options,
+        path,
+        spanStr,
+        spec: jobStatusOperationSpec,
+        tracing,
+      })
+    );
+  };
+}
+
+/**
+ * @internal
+ */
+export function createAnalyzeBatchLro(settings: {
+  client: GeneratedClient;
```

Comment:
nit: extracting this to an interface can help readability

Reply:
I didn't because the type is not being used anywhere else but please let me know if you feel strongly about it.

Reply:
Nope I don't feel strongly about this at all